lib/azure/armrest/configuration.rb

```diff
@@ -198,36 +204,17 @@ def self.log=(output)
       # then a warning will be issued, but no error will be raised.
       #
       def validate_subscription
-        url = File.join(Azure::Armrest::RESOURCE, 'subscriptions') + "?api-version=#{api_version}"
-
-        options = {
-          :url         => url,
-          :proxy       => proxy,
-          :ssl_version => ssl_version,
-          :ssl_verify  => ssl_verify,
-          :headers     => {
-            :accept        => accept,
-            :content_type  => content_type,
-            :authorization => token
-          }
-        }
-
-        response = Azure::Armrest::ArmrestService.send(:rest_get, options)
-        json = JSON.parse(response.body)['value']
-
-        subscriptions = json.map { |hash| [hash['subscriptionId'], hash['state']] }
-
-        found = subscriptions.find { |array| array.first == subscription_id }
+        found = @subscriptions.find { |sub| sub.subscription_id == subscription_id }
 
         unless found
           raise ArgumentError, "Subscription ID '#{subscription_id}' not found"
         end
 
-        if found.last.casecmp("enabled") != 0
-          warn "Subscription '#{found.first}' found but not enabled."
+        if found.state.casecmp('enabled') != 0
+          warn "Subscription '#{found.subscription_id}' found but not enabled."
         end
 
-        true
+        found.subscription_id
```

Comment:
We don't need to purposely return anything because an error will be raised if the validation fails.